### ISSUE TITLE
[Aggregate Function] Support DataSketches in StarRocks

### ISSUE THREAD

**Issue by LiShuMing** [StarRocks/starrocks#50671]
- [x] https://github.com/StarRocks/starrocks/pull/56960
## Enhancement

Since StarRocks has already integrated with [DataSketches](https://datasketches.apache.org/) and supported [HLLSketch](https://github.com/StarRocks/starrocks/issues/20831), but we can support more sketches for more user scenes.

- `ds_theta_count_distinct`(Theta Sketch): https://github.com/StarRocks/starrocks/pull/56960
- `ds_quantile_percentile_approx`(Quantile Sketches): 
- `ds_frequency_approx_top_k`(Frequency Sketches )

- ....

**Comment by chenminghua8** [comment]
@LiShuMing Can you assign this task to me? I want to try it.

**Comment by LiShuMing** [comment]
Of course. Thanks for your contributions. You can do this by referring `ds_hll_count_distinct.h`.

**Comment by yuriolive** [comment]
@cresny started working on it here https://github.com/StarRocks/starrocks/pull/73461
